src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/PooledObjects/PooledArrayBuilder`1.cs

```diff
@@ -67,6 +67,17 @@ public void Add(T item)
         _builder.Add(item);
     }
 
+    public void AddRange(ImmutableArray<T> items)
+    {
+        if (items.Length == 0)
```

Comment:
```suggestion
        if (items.IsDefaultOrEmpty)
```

Reply:
Actually, `ImmutableArray.Builder` doesn't allow a default `ImmutableArray<T>` in `AddRange`. It immediately touches `ImmutableArray.Length`, which will throw:

https://github.com/dotnet/runtime/blob/7d15ceb59982add236b6254b477294a5f3eff112/src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.Builder.cs#L405-L412

IMO, if an uninitialized `ImmutableArray<T>` were passed to this method, it would represent a bug and we probably shouldn't hide it.